### PR TITLE
#169258591 User search functionality

### PR DESCRIPTION
**Pull request by victkarangwa**
#### What does this PR do?
This PR adds search functionality for the users
#### Description of Task to be completed?
- User is able to search a specific record from the requests table
- Admin is able to search for a certain user fro users table
#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend.git`
- `cd team-odd-bn-frontend`
- `git checkout /ft-user-search-functionality-169258591`
- `npm install` packages
- `npm run dev`
- Log in to the system
- Navigate to `Users` if you logged in as a super admin and search for a user by email
- Navigate to `My trips` if you logged in as a normal user
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#169258591](https://www.pivotaltracker.com/story/show/169258591)
#### Screenshots (if appropriate)
Screen Shot 2020-01-30 at 18 50 43

Screen Shot 2020-01-30 at 18 46 12

#### Questions:
N/A

### CODE REVIEW

File: src/__tests__/containers/admin/UserRoles.test.js [hunk 1]
```diff
@@ -74,17 +78,17 @@ describe('User roles test suite', () => {
...
+  //   const component = setUp({});
+  //   const componentDidMountSpy = jest.spyOn(component.instance(), 'componentDidMount');
+  //   component.find('[data-test="userRoles-test"]');
+  //   expect(componentDidMountSpy).not.toHaveBeenCalled();
+  // });
+  // it('Should be able to search', () => {
+  //   const component = setUp({});
+  //   // component.find('[data-test="dataFilter-test"]').simulate('change', 'admin');
+  //   // component.find('[data-test="dataFilter-test"]').simulate('change', 'me');
+  // });
```
Comment (NiyongaboEric) [inline src/__tests__/containers/admin/UserRoles.test.js]: Remember to remove or uncomment lines of code.

File: src/__tests__/helpers/pagination.test.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+describe('Test pagination ', () => {
+  it('handle pagination styles', () => {
+    const cc = PaginationStyle(htmlCollection, 2);
+    expect(cc).toEqual(undefined);
+  });
+  it('test pagination function', () => {
+    const cc = Paginate([1, 2, 3, 4, 5, 6], 2);
+    expect(cc).toEqual([[1, 2], [3, 4], [5, 6]]);
+  });
+});
```
Comment (NiyongaboEric) [inline src/__tests__/helpers/pagination.test.js]: Refactor the test by separating mock data and test.

File: src/helpers/Paginate.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  .map((item, index) => index % itemsPerPage === 0 && allData.slice(index, index + itemsPerPage))
+  .filter(Boolean);
+export const PaginationStyle = (trips, pageNo) => {
+  for (let i = 0; i < trips.length; i++) {
+    trips[i].children[0].style.backgroundColor = '#ffffff';
+    trips[i].children[0].style.color = '#00b9f2';
+  }
+  trips[pageNo].children[0].style.backgroundColor = '#00b9f2';
+  trips[pageNo].children[0].style.color = '#ffffff';
+};
```
Comment (NiyongaboEric) [inline src/helpers/Paginate.js]: Fix EOL issue.

**Review by NiyongaboEric — CHANGES_REQUESTED** [review]

**Review by nziokaivy — CHANGES_REQUESTED** [review]
Great job @victorkarangwa4! I really like the implementation of this feature. I think we can make it better and more user-friendly over time.

However, I have a few concerns
- What happens to a user/manager who would like to search only by origin or destination?
- Also, I think the search fields would be better if they were under the `travel request` tittle and then I think they may look better if they are positioned either tot he left or right and not in the centre.
image

**Review by NiyongaboEric — APPROVED** [review]

File: src/components/PopUp.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import propTypes from 'prop-types';
+
+dotenv.config();
+
+class SubmitButton extends Component {
+  // state={
+  //     display:'none'
+  // }
+  // componentDidMount(){
+  //     this.setState({
```
Comment (hezronkimutai) [inline src/components/PopUp.js]: Remove this comments @victorkarangwa4

File: src/views/trips/RequestsView.js [hunk 5]
```diff
@@ -2,37 +2,80 @@ import React from 'react';
...
+      </select>
+      <input
+        data-test="dataFilter-test"
+        className="search"
+        id="searchInput"
+        type="text"
+        onChange={(e) => {
+          const newData = handleFilter(e, data, search.key);
+          searchResults(newData);
+          // return handleFilter(e, data, search.key);
```
Comment (hezronkimutai) [inline src/views/trips/RequestsView.js]: Remove this comments too.

**Review by hezronkimutai — CHANGES_REQUESTED** [review]
I see you have done a nice work @victorkarangwa4 , but there are still comments which we don't need, please remove them and we will be good to go.

**Review by hezronkimutai — APPROVED** [review]
Nice work.

**Review by nziokaivy — APPROVED** [review]

**Review by shemaeric — APPROVED** [review]

**Pull request merged**